staging/src/k8s.io/apiserver/pkg/storage/interfaces.go

```diff
@@ -251,6 +252,11 @@ type Interface interface {
 	// TODO: Remove when storage.Interface will be separate from etc3.store.
 	// Deprecated: Added temporarily to simplify exposing RequestProgress for watch cache.
 	RequestWatchProgress(ctx context.Context) error
+
+	// Supports checks whether the etcd instance supports the specified feature and returns true if it does,
+	// or false otherwise. It utilizes the provided context.Context for cancellation or timeout control.
+	// The function accepts one featuregate.Feature parameter representing the features to check.
+	Supports(ctx context.Context, feature featuregate.Feature) (bool, error)
```

Comment:
Could we decouple featuregates from storage features? From caller perspective it is unknown how supported featuregates relate to methods on interface and there is no reason for storage know about featuregates. 

ConsistentListFromWatch need a functioning RequestWatchProgress, It will be clearer if caller can just check `storage.Supports(ctx, RequestWatchProgress)`  before calling `storage.RequestWatchProgress`.

Reply:
I missed that initially - I agree with this.

Reply:
Thanks for pointing this out, that was something that i thought yesterday but i didn't know how should i implement it so i put feature gates there. I'll update that